### PR TITLE
Authentication via `Authorization` header

### PR DESCRIPTION
**Pull request by evantahler**
This PR allows the passing of a Grouparoo API Key via header.   Depending on your infrastructure and logging, it is often more secure to pass the API Key via header than via query param. 

Docs @ https://github.com/grouparoo/www.grouparoo.com/pull/881

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

**Review by pedroslopez — COMMENTED** [review]
Question: why not use the standard `Authorization` header for this?

File: core/src/modules/middleware/authentication.ts [hunk 1]
```diff
@@ -218,3 +220,17 @@ async function authenticateTeamMemberInRoom(
...
   });
 }
+
+// check if the apiKey was sent via the header
+function parseHeaders(data: ActionProcessor<any>) {
+  if (data.connection.type !== "web") return;
+
+  if (data.connection.rawConnection.req.headers["x-grouparoo-api-key"]) {
+    data.params.apiKey =
+      data.connection.rawConnection.req.headers["x-grouparoo-api-key"];
```
Comment (edmundito) [inline core/src/modules/middleware/authentication.ts]: My only concern about the implementation here is that you're the data object's parsed `?apiKey=` to assign the value the header, therefore you'd have to understand the implementation of this function to know where the value is really coming from.

The ideal case would be to have the apiKey come from either params or the header itself, but that would take a lot more code changes. Same for the server token below.

**Comment by evantahler** [comment]
> Question: why not use the standard `Authorization` header for this?

Love it.

File: core/src/modules/middleware/authentication.ts (hunk 1 above)
Comment (evantahler) [reply core/src/modules/middleware/authentication.ts]: Well, the goal was to allow for both - the user can choose to send the key and server token either via header or query param - with the header winning if they are both set

**Comment by edmundito** [comment]
Are we using the Authorization spec correctly here?
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization

**Comment by evantahler** [comment]
Talking it over with @pedroslopez, We are going with type=Bearer

https://swagger.io/docs/specification/authentication/bearer-authentication/

> Bearer authentication (also called token authentication) is an HTTP authentication scheme that involves security tokens called bearer tokens. The name “Bearer authentication” can be understood as “give access to the bearer of this token.” The bearer token is a cryptic string, usually generated by the server in response to a login request. The client must send this token in the Authorization header when making requests to protected resources:

**Comment by evantahler** [comment]
Now with proper use of `Authentication: Bearer <token>` headers!

**Review by pedroslopez — APPROVED** [review]
Nice!

**Pull request merged**